### PR TITLE
LibJS: Add class fields and other small fixes

### PR DESCRIPTION
**Pull request by davidot**
Was working on some class fixes and implemented fields, thought just private fields was ecma 2022 spec but public fields is as well. Whoops.
Example:
```javascript
class A {
  value = 3

  static also_possible = true;
}

new A().value === 3
```

We still do a small number of things wrong like `NamedEvaluation` things but that is a more general issue so I just left that.

### CODE REVIEW

File: Userland/Libraries/LibJS/Parser.cpp [hunk 1]
```diff
@@ -365,7 +365,8 @@ NonnullRefPtr<Declaration> Parser::parse_declaration()
 NonnullRefPtr<Statement> Parser::parse_statement(AllowLabelledFunction allow_labelled_function)
```
Comment (davidot) [inline Userland/Libraries/LibJS/Parser.cpp:365]: This commit got kind of big but I didn't want to litter with a bunch of tiny commits. I can split them up if needed

File: Userland/Libraries/LibJS/Parser.cpp [hunk 2]
```diff
@@ -2503,9 +2514,15 @@ NonnullRefPtr<IfStatement> Parser::parse_if_statement()
         // Code matching this production is processed as if each matching occurrence of
         // FunctionDeclaration[?Yield, ?Await, ~Default] was the sole StatementListItem
         // of a BlockStatement occupying that position in the source code.
+        VERIFY(match(TokenType::Function));
         ScopePusher scope(*this, ScopePusher::Let, Parser::Scope::Block);
         auto block = create_ast_node<BlockStatement>({ m_state.current_token.filename(), rule_start.position(), position() });
-        block->append(parse_declaration());
+        auto function_declaration = parse_declaration();
+        VERIFY(is<FunctionDeclaration>(*function_declaration));
+        auto& func_decl = static_cast<FunctionDeclaration const&>(*function_declaration);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Parser.cpp:2547]: as mentioned in previous PRs, please avoid abbreviations as much as possible - there's no reason to not spell out `function_declaration` :)

**Review by linusg — COMMENTED** [review]
Typo in first commit message:

> Fix that expressions statement allowed too much.

**Comment by davidot** [comment]
Fixed the commit message and made the name more expressive.

File: Userland/Libraries/LibJS/Parser.cpp (hunk 2 above)
Comment (davidot) [reply Userland/Libraries/LibJS/Parser.cpp]: This habit seems to be quite hard to kick for me, thanks for the good eye.

**Review by linusg — APPROVED** [review]

**Pull request merged**